:book: Update the README (#2823)
kubernetes-sigs/kubebuilder: README.md

developers for rapidly building and publishing Kubernetes APIs in Go.
It builds on top of the canonical techniques used to build the core Kubernetes APIs to provide simple abstractions that reduce boilerplate and toil.

Kubebuilder does **not** exist as an example to *copy-paste*, but instead provides powerful libraries and tools
to simplify building and publishing Kubernetes APIs from scratch. On to of that, it 
provides an plugin architecture which allows users take advantage of optional helpers
and features. To know more about see the [Plugin section][plugin-section].

Kubebuilder is developed on top of the [controller-runtime][controller-runtime] and [controller-tools][controller-tools] libraries.

### Kubebuilder is also a framework for other tools

Kubebuilder is extensible and can be imported to be used as a library in other projects.
Specifically, for those which are looking for a way to use Kubebuilder's existing CLI and scaffolding the projects and re-use its plugins, 
but to also be able to augment the Kubebuilder project structure with other custom project types so that it is possible to support the Kubebuilder 
workflow with non-Go operators. 

> A good example of its usage is the project [Operator-SDK][operator-sdk]
> that uses Kubebuilder as lib to generate the scaffolds and create its own helper plugins 
> to provide additional features and options on top. [Operator-SDK][operator-sdk] also uses Kubebuilder
> to support their workflow with non-Go operators (_e.g. operator-sdk's Ansible and Helm-based language Operators_)

Reviewer: ```suggestion
[Operator-SDK][operator-sdk] is a good example of a project that uses Kubebuilder as a library. [Operator-SDK][operator-sdk] uses the plugin feature to include non-Go operators _e.g. operator-sdk's Ansible and Helm-based language Operators_.
```

> To know more about see [how to create your own plugins][your-own-plugins]. 

Reviewer: ```suggestion
To learn more see [how to create your own plugins][your-own-plugins]. 
```

## Supportability

Currently, Kubebuilder officially supports OSX and Linux platforms. 
So, if you are using a Windows OS you may find issues. Contributions towards 
supporting Windows are welcome.

### Apple Silicon

> The current scaffold done by the CLI (`go/v3`) when we run `kubebuilder init` uses [kubernetes-sigs/kustomize][kustomize] v3 which does not provide
> a valid binary for Apple Silicon (`darwin/arm64`). Therefore, you can use the `go/v4-alpha` plugin
> instead which provides support for this platform:

Reviewer: You could simplify this and just let them know that M1 is supported using v4.

```suggestion
Apple Silicon (`darwin/arm64`) is supported using the `go/v4-alpha` plugin which provides support for this platform.
```

If you must let them know why it is broken, then link to an issue. But having the full explanation as to why it doesn't work just makes it wordy.

```suggestion
Apple Silicon (`darwin/arm64`) is supported using the `go/v4-alpha` plugin which provides support for this platform. There is an [issue](https://github.com/kubernetes-sigs/kubebuilder/issues/2566) with the current `go/v3` plugin and Apple Silicon.
```

Author: @jmrodri it is a good point. Changing
